### ISSUE TITLE
VS Code VARIABLES View does not show DFDL Variables

### ISSUE THREAD

**Issue by ds-nteligen** [apache/daffodil-vscode#1092]
{Sev: minor}

The VS Code view called VARIABLES does not show DFDL variables. This could be perceived as misleading when working with DFDL files.

**Comment by lrbarber** [comment]
Does the Daffodil internal debugger provide this information for external consumption?
If not, is there a way to remove the variables display area from VS Code?

**Comment by stevedlawrence** [comment]
> Does the Daffodil internal debugger provide this information for external consumption?

Yes, the debugger has access to `ParserOrUnparseState`, which has a `variableMapForDebugger` member that contains information about all the variables (both current and previous instances). The CLI "info variables" debugger command has an example of how to access this state information:

https://github.com/apache/daffodil/blob/main/daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/debugger/InteractiveDebugger.scala#L1939-L1989

Note that it only displays the current instance of each variable--an enhancement might be to give the ability to display all instances instead.

**Comment by michael-hoke** [comment]
This is being done. If you use a schema that has variables in it, they will show up under Schema, grouped by the namespace that the variables are in. No_Namespace will show up if there isn't one.
